x/mongo/driver/topology/server_options.go

```diff
@@ -55,9 +56,17 @@ func newServerConfig(opts ...ServerOption) *serverConfig {
 		opt(cfg)
 	}
 
+	cfg.sanityCheck()
 	return cfg
 }
 
+func (cfg *serverConfig) sanityCheck() {
+	const tmpl = "x/mongo/driver/topology: internal error: serverConfig should be: %s"
```

Comment:
Hmm I don't love the `panic` here. I think generally, we want to return an informative error to the user as soon as possible as opposed to `panic`ing deeper within the `x` package. I'm wondering if we can set the error on the returned `ClientOptions` object in [`ApplyURI`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/clientoptions.go#L236) instead. Something like:

```
// Verify that if both minPoolSize and maxPoolSize are set, minPoolSize < maxPoolSize.
if cs.MinPoolSizeSet && cs.MaxPoolSizeSet && cs.MinPoolSize >= cs.MaxPoolSize {
  c.err = fmt.Errorf("set minPoolSize %d exceeds set maxPoolSize %d", cs.MinPoolSize, cs.MaxPoolSize)
  return c
}
```